pinot-core/src/main/java/org/apache/pinot/core/data/manager/InstanceDataManager.java

```diff
@@ -117,19 +118,19 @@ void deleteSegment(String tableNameWithType, String segmentName)
    * Download happens when local segment's CRC mismatches the one of the remote segment; but can also be forced to do
    * regardless of CRC.
    */
-  void reloadSegment(String tableNameWithType, String segmentName, boolean forceDownload)
+  void reloadSegment(String tableNameWithType, String segmentName, boolean forceDownload, String reloadJobId)
       throws Exception;
 
   /**
    * Reloads all segments of a table.
    */
-  void reloadAllSegments(String tableNameWithType, boolean forceDownload)
+  void reloadAllSegments(String tableNameWithType, boolean forceDownload, String reloadJobId)
       throws Exception;
 
   /**
    * Reload a list of segments in a table.
    */
-  void reloadSegments(String tableNameWithType, List<String> segmentNames, boolean forceDownload)
+  void reloadSegments(String tableNameWithType, List<String> segmentNames, boolean forceDownload, String reloadJobId)
```

Comment:
nit: For these function, should we keep the original versions which pass `reloadJobId` as null and add new versions of the API to take the `reloadJobId`? This can help reduce some changes where we just pass null.